examples/hybrid/schur_complement_W.jl

```diff
@@ -206,3 +198,82 @@ function linsolve!(::Type{Val{:init}}, f, u0; kwargs...)
         end
     end
 end
+
+"""
+    thomas_algorithm!(A, b)
+
+Thomas algorithm for solving a linear system A x = b,
+where A is a tri-diagonal matrix.
+A and b are overwritten.
+Solution is written to b
+"""
+function thomas_algorithm!(A, b)
+    # A[_, 1] is lower diag, A[_, 2] is main diag, A[_, 3] is upper diag
+    nrows = size(A, 1)
+    # first row
+    A[1, 3] /= A[1, 2]
```

Comment:
Accessing by `A[row, diag]`, is a bit confusing. Instead of accessing by `parent`, can we instead use `getidx` on the `Field`?

@dennisYatunin can you help with this?

Reply:
Since the matrix is represented by a field of matrix rows (`StencilCoefs`), we do have that `parent(A)[row, diag] == getidx(A, row, ...)[diag]`. However, there isn't a good way to use `setidx` to set an individual entry on a row; we can only use it to set all the entries in a row at once. This algorithm, as it is written, uses the third diagonal of `A` as a cache for computing `A \ b`, so it isn't really compatible with `setidx`. However, we could also allocate a cache instead of overwriting the third diagonal of `A`, and then use `setidx` to set the entries in that cache (each entry would just be a floating-point value). Would that be a good alternative?

Reply:
Also, we probably want to avoid modifying `A` in this function, because we might want to reuse the same Jacobian matrix for multiple Newton iterations. I think allocating a cache and leaving the third diagonal of `A` unchanged would be better than what we have here. Note that the current code doesn't modify the original matrix field; instead, it copies the field data into an array, and then modifies the third diagonal in that array.

Reply:
No, I would prefer that we don't allocate a cache for now: the extra data movement will start to get expensive. If we need it, we can add it as an option.

Reply:
Can we clarify this: can we write this as `thomas_algorithm!(A, b, cache)`? so that `thomas_algorithm!` is non-allocating, but we have additional storage for handling things more intuitively?

Reply:
Also, I'm not a fan of overwriting `b` with `x`. It's very awkward from a user POV. We have to compute tendencies, pass them in, and suddenly it becomes the state vector. If, for some reason, we want to keep the tendencies around (say for computing the residual, or for limiters or something), then this function corrupts that data. One extra field is not a performance issue here.

Reply:
This isn't trying to be a generic `ldiv!` (which is why I gave it a different name).

Look, there are lots of ways we can do this. For now, just make it completely in-place, with no cache. That solves the immediate problem in the simplest way possible. If we decide we need something else, we can do that then.

Reply:
As for the `getidx`/`setidx!`, you can define
```
ld = S.coefs.:1
d = S.coefs.:2
ud = S.coefs.:3
```
and operate on those